### PR TITLE
Fixes #31670 - don't timeout when running db:migrate

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

File: manifests/database.pp [hunk 1]
```diff
@@ -22,7 +22,8 @@
     }
 
     foreman::rake { 'db:migrate':
-      unless => '/usr/sbin/foreman-rake db:abort_if_pending_migrations',
+      timeout => 0,
```
Comment (evgeni) [inline manifests/database.pp:25]: I am not 100% sure 0 is a good value here, but couldn't think of anything else.

Comment (ekohl) [reply manifests/database.pp:25]: You could make it a parameter on the (private) class so you at least have a way to change it as a local workaround.

Comment (evgeni) [reply manifests/database.pp:25]: You mean a param of the `foreman::database` class?

Comment (ekohl) [reply manifests/database.pp:25]: Exactly.

Comment (evgeni) [reply manifests/database.pp]: So, like this?

**Review by ehelms — APPROVED** [review]

File: manifests/database.pp [hunk 2]
```diff
@@ -1,5 +1,8 @@
-# Set up the foreman database
-class foreman::database {
+# @summary Set up the foreman database
+# @api private
+class foreman::database(
+  Integer $timeout = 0,
```
Comment (ekohl) [inline manifests/database.pp:4]: You can set a lower bound (and upper bound, but I don't think that makes sense here)
```suggestion
  Integer[0] $timeout = 0,
```

**Review by ekohl — APPROVED** [review]

File: manifests/database.pp (hunk 2 above)
Comment (evgeni) [reply manifests/database.pp]: done

**Comment by evgeni** [comment]
I've also bumped the timeout to 3600 seconds (1h), as discussed earlier.

@ehelms @ekohl ACk still holds?

**Review by ekohl — APPROVED** [review]

**Pull request merged**